controller_manager/src/controller_manager.cpp

```diff
@@ -1745,6 +1746,16 @@ void ControllerManager::read(const rclcpp::Time & time, const rclcpp::Duration &
     deactivate_controllers(rt_controller_list, stop_request);
     // TODO(destogl): do auto-start of broadcasters
   }
+  else if (failed_hardware_names.size() > 0)
+  {
+    // Status is ok but some hardware is not ok (SKIPPED)
+    // Determine controllers to skip
+    for (const auto & hardware_name : failed_hardware_names)
+    {
+      auto controllers = resource_manager_->get_cached_controllers_to_hardware(hardware_name);
+      controllers_to_skip.insert(controllers_to_skip.end(), controllers.begin(), controllers.end());
+    }
+  }
```

Comment:
I don't think this is needed here. Hardware interface is not failed. The issue here is that we are calling `read` and `write` from HW interfaces constantly, and in some cases we skip it (when not `inactive` or `active`). But then the controllers cannot run at all since they don't have interfaces to connect to – so I doubt that we have to do anything with them here.